src/File.php

```diff
@@ -485,6 +489,56 @@ public function canDelete($member = null)
             ->canDelete($this->ID, $member);
     }
 
+    /**
+     * Check if the File is viewable to a public user i.e. someone who is not logged in to the CMS
+     */
+    public function canViewAnonymous(): bool
+    {
+        return Member::actAs(null, function(): bool {
+            return $this->canView();
+        });
+    }
+
+    /**
+     * Whether the file was originally uploaded in a userform submissin on a FileFiled
+     */
+    public function isUserDefinedFormUpload(): bool
```

Comment:
I think this would make more sense as an Extension applied from the userforms module. I would rather avoid tying core module with non-core module if possible.